passes/opt/opt_expr.cc

```diff
@@ -259,6 +259,20 @@ bool is_one_or_minus_one(const Const &value, bool is_signed, bool &is_negative)
 	return last_bit_one;
 }
 
+int get_highest_hot_index(RTLIL::SigSpec signal)
+{
+	for (int i = GetSize(signal) - 1; i >= 0; i--)
+	{
+		if (signal[i] == RTLIL::State::S0)
+			continue;
+
+		if (signal[i] == RTLIL::State::S1)
+			return i;
+	}
```

Comment:
I think I wanted a `break` here or something.

Reply:
Oh, yeah, that would make sense then. (I have to admit I haven't had a close look at the users of `get_highest_hot_index()`. So I don't really know which behavior makes sense, just noticed that the first `if ()` statement didn't contribute to the behavior of the function.)

Reply:
Fixed. The handling of x should be conservative...